pygmt/datasets/earth_deflection.py

```diff
@@ -0,0 +1,117 @@
+"""
+Function to download the IGPP Earth east-west and south-north deflection datasets from
+the GMT data server, and load as :class:`xarray.DataArray`.
+
+The grids are available in various resolutions.
+"""
+
+from collections.abc import Sequence
+from typing import Literal
+
+import xarray as xr
+from pygmt.datasets.load_remote_dataset import _load_remote_dataset
+
+__doctest_skip__ = ["load_earth_deflection"]
+
+
+def load_earth_deflection(
+    resolution: Literal[
+        "01d", "30m", "20m", "15m", "10m", "06m", "05m", "04m", "03m", "02m", "01m"
+    ] = "01d",
+    region: Sequence[float] | str | None = None,
+    registration: Literal["gridline", "pixel", None] = None,
+    direction: Literal["edefl", "ndefl"] = "edefl",
```

Comment:
If people have a more suitable name for this new parameter, feel free to make suggestions. I just picked the "east-west deflection" dataset as default.

Reply:
I think the parameter is OK, but for the arguments, "edefl"/"ndefl" is not readable. Maybe "e"/"n" or "east"/"north" or "east_west"/"north_south", but I have to admit that I have no idea about the physical meaning of this dataset

Reply:
See comment https://github.com/GenericMappingTools/pygmt/pull/3728#discussion_r1898876704.